### ISSUE TITLE
PhoneNumberKit dependency update

### ISSUE THREAD

**Issue by barniques** [checkout/frames-ios#382]
Greetings!

We need to update PhoneNumberKit.

In the current version 3.3.3, `PhoneNumberKit` uses the deprecated `isoCountryCode` method. The comment states that the method may return `'--'` instead of the region code, and unfortunately, we have started to encounter such cases. `PhoneNumberKit` fixed this issue in version 3.4.3.

However, the current version of `Frames` strictly specifies version 3.3.3 in podspec, which prevents us from updating.

The solution I propose is:
- update the minor version of PhoneNumberKit 
- specify the dependency version in the podspec as `'~> 3.4.3'`, witch will allow patch versions with bug fixes to be raised without any problems.

**Comment by alex-ioja-yang-cko** [comment]
Hi @barniques,

Thank you for flagging this to us. You are correct, PhoneNumberKit has reached 3.5.8 and our CocoaPods distribution is lagging behind by having fixed 3.3.3.

We will discuss this internally and I am confident we will aim to improve soon. But as these processes may take a longer time, I have looked up the SwiftPackageManager distribution and importing it that way resolves the dependency locally to the latest 3.5.8.

To avoid being blocked by our release, would trying Swift Package Manager for the dependency be of any help to your team? We have tried SPM over many different setups and it should work seamlessly alongside CocoaPods so there should be no negative impact on your project or other dependencies.

If you could please let us know whether this may be an immediate solution it for your team and we will aim to come back with an update once we review how we may fit this change on the product development.

Have a great evening,

Alex

**Comment by alex-ioja-yang-cko** [comment]
Hi @barniques,

A small update from our side.

Whilst working on a new feature that will also allow us to update the dependency, we did find that PhoneNumberKit latest Cocoapods release is actually 3.3.4, from last year (as found in  [Cocoapods repo](https://github.com/CocoaPods/Specs/blob/master/Specs/2/4/b/PhoneNumberKit/3.3.4/PhoneNumberKit.podspec.json)).

This means that whilst in an upcoming release we will be removing the "fixed version dependency" as you suggested, simply relying on Cocoapods to get the correct version may not work (although I am not familiar with Cocoapods & SPM ability to interlink).

There is however an [issue opened on PhoneNumberKit](https://github.com/marmelroy/PhoneNumberKit/issues/639) that is also requesting a new release to Cocoapods.

In the meanwhile, if SPM has helped fix your issue, happy to hear any feedback,

Thank you,

Alex

**Comment by nberthelot** [comment]
Hi @alex-ioja-yang-cko ,

Thank you for your update do you have an ETA about the fix ?

Thank you

**Comment by alex-ioja-yang-cko** [comment]
Hi @nberthelot,

I'm afraid I wouldn't be able to offer an ETA right now, as any release will be subject to Product and has implications beyond the setup we are looking at here. It would be best to reach out to your company's operations contact for estimates.

However, given the Cocoapods spec update would only unlock PhoneNumberKit via Cocoapods to `3.3.4`, is this still a blocker for you? It looks that the latest versions are still not available outside of SPM. And using Frames via SPM will already use these later versions regardless.

Let me know if there are other complexities or info I am missing, as it might help us to scope and ensure its included in upcoming release.

Thank you,

Alex

**Comment by barniques** [comment]
Hi @alex-ioja-yang-cko and @nberthelot,

At the moment, we have solved the problem through our private repo for the spec repo.
In our project, there are dependencies that are no longer supported by CocoaPods. With the help of our private spec repo, we can add support ourselves.
Regarding PhoneNumberKit, we took a subspec for version 3.5.8 and changed the version to 3.3.3. This allowed us to use the latest version of PhoneNumberKit and get rid of the bug.
Our team understands that this is a temporary solution, but we currently don't have the resources to move to SPM, although we would very much like to.
I think in 1-2 months, we will fully switch to SPM, which will eliminate problems like this.

In any case, thank you for your help,

Dmitry
